eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientInstantiationCheck.java

```diff
@@ -181,14 +201,206 @@ private void checkClassField(DetailAST objBlockToken) {
+     *    Asynchronous method with annotation @ServiceMethod must not has {@code Context} as a parameter.
+     *
+     * @param methodDefToken METHOD_DEF AST node
+     */
+    private void checkMethodNamingPattern(DetailAST methodDefToken) {
+        DetailAST modifiersToken = methodDefToken.findFirstToken(TokenTypes.MODIFIERS);
+        DetailAST serviceMethodAnnotation = hasServiceMethodAnnotation(modifiersToken);
+        // NOT a @ServiceMethod method
+        if (serviceMethodAnnotation == null) {
+            return;
+        }
+
+        String methodName = methodDefToken.findFirstToken(TokenTypes.IDENT).getText();
+
+        // 1) Follows method naming pattern. Refer to Java Spec.
+        if (!isCommonNamingPattern(methodName)) {
+            log(methodDefToken, String.format("Method name ''%s'' should follow a common vocabulary. Refer to Java Spec: %s.",
+                methodName, JAVA_SPEC_LINK));
+        }
+
+        // 2) Methods should not have "Async" added to the method name
+        if (methodName.contains(ASYNC)) {
+            log(methodDefToken, String.format("Method name ''%s'' should not contain ''%s'' in the method name.",
+                methodName, ASYNC));
+        }
+
+        // 3) The return type of async and sync clients should be as per guidelines
+        checkServiceClientMethodReturnType(methodDefToken, serviceMethodAnnotation);
+
+        // 4) Check 'withResponse' naming pattern
+        checkReturnTypeNamingPattern(methodDefToken, methodName);
+
+        // 5) Synchronous method with annotation @ServiceMethod has to have {@code Context} as a parameter.
+        // Asynchronous method with annotation @ServiceMethod must not has {@code Context} as a parameter.
+        checkContextInRightPlace(methodDefToken);
+    }
+
+    /**
+     * Checks for the return type of async and sync clients should be as per guidelines:
+     *     1) The return type for async collection should be of type? extends Flux
+     *     2) The return type for async single value should be of type? extends Mono
+     *     3) The return type for sync collection should be of type? extends Stream
+     *     4) The return type for sync single value should be of type? extends Response
+     *
+     * @param methodDefToken METHOD_DEF AST node
+     * @param serviceMethodAnnotation ANNOTATION AST node which used to find the if the annotation has 'return' key,
+     * if found. return the value of member'return'.
+     */
+    private void checkServiceClientMethodReturnType(DetailAST methodDefToken, DetailAST serviceMethodAnnotation) {
+        // Find the annotation member 'returns' value
+        String returnsAnnotationMemberValue = getAnnotationMemberReturnsValue(serviceMethodAnnotation, "returns");
+
+        StringBuilder sb = new StringBuilder();
+        getReturnType(methodDefToken.findFirstToken(TokenTypes.TYPE), sb);
+        String returnType = sb.toString();
```

Comment:
create a new method where `StringBuilder sb` is created and where `getReturnType` is called from and called directly like `String returnType = getReturnType(methodDefToken)`  
That way we don't duplicate the logic of creating the StringBuilder and calling the first Type token many times where we want the return type.